tests/core_adapter/test_adapter_database.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+import logging
+from sqlalchemy.orm import Session
+from pyramid_oereb.core.adapter import DatabaseAdapter
+
+
+def test_init():
+    adapter = DatabaseAdapter()
+    assert isinstance(adapter._connections_, dict)
+
+
+def test_get_connections_instance():
+    adapter = DatabaseAdapter()
+    assert isinstance(adapter.get_connections(), dict)
+
+
+def test_add_connection():
+    db_url = 'postgresql://adapter_user:adapter_pswd@adapter-tests-00:5432/adapter-db'
+    adapter = DatabaseAdapter()
+    adapter.add_connection(db_url)
+    assert isinstance(adapter.get_session(db_url), Session)
+
+
+def test_add_connection_already_exists(caplog):
+    caplog.set_level(logging.INFO)
+    db_url = 'postgresql://adapter_user:adapter_pswd@adapter-tests-01:5432/adapter-db'
+    adapter = DatabaseAdapter()
+    adapter.add_connection(db_url)
+    assert 'Connection already exists: {0}'.format(db_url) not in caplog.text
+    adapter.add_connection(db_url)
+    assert 'Connection already exists: {0}'.format(db_url) in caplog.text
+
+
+def test_get_connections():
+    db_url = 'postgresql://adapter_user:adapter_pswd@adapter-tests-02:5432/adapter-db'
+    test_connection = {db_url: {"engine": "test_engne", "session": "test_session"}}
```

Comment:
small typo: test_engne instead of test_engine - not an issue for the test

Reply:
Corrected.